src/server/session.ts

```diff
@@ -967,7 +967,22 @@ namespace ts.server {
             },
             [CommandNames.Open]: (request: protocol.Request) => {
                 const openArgs = <protocol.OpenRequestArgs>request.arguments;
-                this.openClientFile(openArgs.file, openArgs.fileContent);
+                let scriptKind: ScriptKind;
+                switch (openArgs.scriptKindName) {
+                    case "TS":
+                        scriptKind = ScriptKind.TS;
+                        break;
+                    case "JS":
+                        scriptKind = ScriptKind.JS;
+                        break;
+                    case "TSX":
+                        scriptKind = ScriptKind.TSX;
+                        break;
+                    case "JSX":
+                        scriptKind = ScriptKind.JSX;
+                        break;

```

Comment:
Shouldn't this be like `scriptKind = ScriptKind[openArgs.scriptKindName]` ?

I mean how many places will you need to touch if you create a new `ScriptKind`? You want to make it less error-prone.